framework/include/problems/SubProblem.h

```diff
@@ -68,6 +68,80 @@ class SubProblem : public Problem
 
   virtual bool isTransient() const = 0;
 
+  /**
+   * Create a Tag.  Tags can be associated with Vectors and Matrices and allow objects
+   * (such as Kernels) to arbitrarily contribute values to any set of vectors/matrics
+   *
+   * Note: If the tag is already present then this will simply return the TagID of that Tag
+   *
+   * @param tag_name The name of the tag to create, the TagID will get automatically generated
+   */
+  virtual TagID addVectorTag(TagName tag_name);
+
+  /**
+   * Get a TagID from a TagName.
+   */
+  virtual TagID getVectorTagID(const TagName & tag_name);
+
+  /**
+   * Retrieve the name associated with a TagID
+   */
+  virtual TagName vectorTagName(TagID tag);
+
+  virtual std::map<TagName, TagID> & getVectorTag() { return _vector_tag_name_to_tag_id; }
```

Comment:
Should this function name be plural since it can return info for several tags?  It confused me at first at call locations.